snippet/snippet.go

```diff
@@ -30,45 +30,53 @@ func (snippets *Snippets) Load(includeDirs bool) error {
 	// Create a list of snippet files to load snippets from
 	var snippetFiles []string
 
-	// Load snippets from the main snippet file
-	// Raise an error if the file is not found / not configured
-	snippetFile := config.Conf.General.SnippetFile
-	if snippetFile != "" {
+	snippetFile, err := config.ExpandPath(config.Conf.General.SnippetFile)
+	if err == nil {
 		if _, err := os.Stat(snippetFile); err == nil {
-			snippetFiles = append(snippetFiles, snippetFile)
+			snippetFiles = append(snippetFiles, config.Conf.General.SnippetFile)
```

Comment:
Why do you append the un-expanded path here instead of the expanded one? Not clear to me

Reply:
Good point.
They are also expanded in line 69 below: https://github.com/csessh/pet/blob/5b28fbbb1961049d744b4e7ccd9d94e655e4b91c/snippet/snippet.go#L69

Reply:
Ah I see. Hmm. There's a lot of expansions everywhere, sometimes maybe double expansions also. We should check that expansion only happens at the point where it's necessary, i.e. maybe only at line 69 here.

Are the expansions before that line necessary?